### PR TITLE
Fix logic error in loading_screen.rs

### LINKED ISSUE #16792: Error in loading_screen.rs example
## Bevy version

v0.15.0

## What went wrong

I think there is an error in the [example](https://github.com/bevyengine/bevy/blob/latest/examples/games/loading_screen.rs#L220) on line 220.
If there are two elements in the array and the zero element is deleted on the first iteration, then on the second iteration an attempt to delete the element with index 1 will end with a panic because the deletion of the zero element caused a shift.

### PR DESCRIPTION
**Pull request by BenjaminBrienen**
# Objective

Fix #16792

## Solution

Fix the logic to retain loaded ones

## Testing

Unable to test due to #16988

### CODE REVIEW

**Comment by MiniaczQ** [comment]
Example appears to work correctly

**Review by alice-i-cecile — APPROVED** [review]

File: examples/games/loading_screen.rs [hunk 1]
```diff
@@ -204,21 +204,13 @@ fn update_loading_data(
         // we reset the confirmation frame count.
         loading_data.confirmation_frames_count = 0;
 
-        // Go through each asset and verify their load states.
-        // Any assets that are loaded are then added to the pop list for later removal.
-        let mut pop_list: Vec<usize> = Vec::new();
-        for (index, asset) in loading_data.loading_assets.iter().enumerate() {
+        loading_data.loading_assets.retain(|asset| {
             if let Some(state) = asset_server.get_load_states(asset) {
```
Comment (rparrett) [inline examples/games/loading_screen.rs:208]: nit: could be
```rust
asset_server
    .get_recursive_dependency_load_state(asset)
    .map_or(true, |state| !state.is_loaded())
```

Comment (BenjaminBrienen) [reply examples/games/loading_screen.rs:208]: Actually, I think that would trigger a lint for `is_none_or`. :p

Comment (rparrett) [reply examples/games/loading_screen.rs:208]: It doesn't but that is even nicer :)

**Pull request merged**